manifests/params.pp

```diff
@@ -35,6 +35,13 @@
             $package_ensure        = $::mongodb::globals::version
             $package_ensure_client = $::mongodb::globals::version
             $package_ensure_mongos = $::mongodb::globals::version
+            if(versioncmp($::mongodb::globals::version, '2.6.7') >= 0) {
+              $config        = '/etc/mongod.conf'
+              $mongos_config = '/etc/mongos.conf'
+            } else {
+              $config        = '/etc/mongodb.conf'
+              $mongos_config = '/etc/mongodb-shard.conf'
+            }
           } else {
             $server_package_name   = pick($::mongodb::globals::server_package_name, 'mongodb-10gen')
```

Comment:
$config not set here, so there are scenarios that could cause issues. https://github.com/puppetlabs/puppetlabs-mongodb/pull/203 seems to be a better solution.